cnf-tests/testsuites/e2esuite/sro/sro.go

```diff
@@ -0,0 +1,437 @@
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func createSpecialResource() (*srov1beta1.SpecialResource, string, error) {
+	nodesObjList := &corev1.NodeList{}
+	err := client.Client.List(ctx.TODO(), nodesObjList)
+	if err != nil {
+		return nil, "", err
+	}
+	nodesList := make([]string, len(nodesObjList.Items))
+	for idx, node := range nodesObjList.Items {
+		nodesList[idx] = node.Name
+	}
+
+	nn, err := nodes.MatchingOptionalSelectorByName(nodesList)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if len(nn) == 0 {
+		return nil, "", fmt.Errorf("0 nodes match the selector")
+	}
+
+	kernels := make(map[string]bool)
+	kernel := ""
+	for _, nodeName := range nn {
+		node, err := client.Client.Nodes().Get(ctx.TODO(), nodeName, metav1.GetOptions{})
+		if err != nil {
+			return nil, "", err
+		}
+
+		kernels[node.Status.NodeInfo.KernelVersion] = true
+		kernel = node.Status.NodeInfo.KernelVersion
+	}
+
+	if len(kernels) != 1 {
+		return nil, "", fmt.Errorf("multiple kernel versions exist")
+	}
+
+	sp := &srov1beta1.SpecialResource{
+		ObjectMeta: metav1.ObjectMeta{Name: specialResourceName, Namespace: namespaces.SroTestNamespace},
+		Spec: srov1beta1.SpecialResourceSpec{
+			Namespace: namespaces.SroTestNamespace,
+			Chart: helmerv1beta1.HelmChart{
+				Name:    specialResourceName,
+				Version: "0.0.1",
+				Repository: helmerv1beta1.HelmRepo{
+					Name: "chart",
+					URL:  fmt.Sprintf("cm://%s/%s", namespaces.SroTestNamespace, chartName),
+				},
+			},
+			Set: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind":             "Values",
+					"apiVersion":       "sro.openshift.io/v1beta1",
+					"kmodNames":        []string{"ice"},
+					"containerName":    ootDriverImageName,
+					"externalRegistry": "",
+					"signDriver":       false,
+					"downloadDriver":   true,
+					"kernelVersion":    kernel,
+					"buildArgs": []buildArgs{
+						{Name: "KMODVER", Value: "SRO"},
+						{Name: "IMAGE", Value: driverToolKitImage},
+						{Name: "KERNEL_SOURCE", Value: "yum"},
+						{Name: "ICE_DRIVER_VERSION", Value: "1.6.4"},
+					},
+				},
+			},
+			DriverContainer: srov1beta1.SpecialResourceDriverContainer{
+				Artifacts: srov1beta1.SpecialResourceArtifacts{
+					Images: []srov1beta1.SpecialResourceImages{
+						{
+							Name:      fmt.Sprintf("%s:latest", sourceImageStreamName),
+							Kind:      "ImageStreamTag",
+							Namespace: namespaces.SroTestNamespace,
+							Paths: []srov1beta1.SpecialResourcePaths{
+								{
+									SourcePath:     "/usr/src/oot-driver/.",
+									DestinationDir: "./",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err = client.Client.Create(ctx.TODO(), sp)
+
+	return sp, kernel, err
+}
+
+// This will remove all the cluster scope level objects
+func Clean() {
+	specialResourceList := &srov1beta1.SpecialResourceList{}
+	err := client.Client.List(ctx.TODO(), specialResourceList)
```

Comment:
Filter by namespace: namespaces.SroTestNamespace?
All the resources in the test seem to be namespaced.

Reply:
The specialResource CR is cluster wide not namespaced